src/mono/mono/mini/aot-runtime.c

```diff
@@ -2413,6 +2413,59 @@ mono_aot_register_module (gpointer *aot_info)
 		mono_aot_unlock ();
 }
 
+static GHashTable *bundled_assemblies = NULL;
+
+static int bundles_added = 0;
+
+void
+mono_add_bundled_assembly (const char *name, const unsigned char *data, unsigned int size)
```

Comment:
I think this should be more generic resource bundling API and that we should place it somewhere else so it is not AOT only specific feature, current mono_register_bundled_assemblies is for example in assembly.c.

Something like mono_add_bundled_resource and then we can have an enum of different kinds of resources types that we can bundle and probably a generic one as well.

Reply:
Leveraging an enum to classify the types of resources we support bundling (Blob = default, Assembly (.dll), Satellite Assembly (.resources.dll), Binary (.bin)), I am leveraging a `BundledResource` struct that will accommodate all of these types of resources by holding `name` (Assembly/Satellite Assembly), `culture` (Satellite Assembly), `data` (all), `size` (all), `type` (depicts resource type). I opted to use one hashtable overall (instead of one hashtable per resource as that felt like more repetitive code) and each resource is added.

And for registering assemblies and satellite assemblies, `mono_register_bundled_resources` is responsible for invoking each (as long as they have had assemblies added, which is tracked by int array `num_bundled_resources_to_register`)